presto-main/src/test/java/com/facebook/presto/type/TestCharacterStringCasts.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.operator.scalar.AbstractTestFunctions;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.operator.scalar.CharacterStringCasts.varcharToCharSaturatedFloorCast;
+import static com.facebook.presto.spi.type.CharType.createCharType;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static io.airlift.slice.Slices.utf8Slice;
+import static org.testng.Assert.assertEquals;
+
+public class TestCharacterStringCasts
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testVarcharToVarcharCast()
+            throws Exception
+    {
+        assertFunction("cast('bar' as varchar(20))", createVarcharType(20), "bar");
+        assertFunction("cast(cast('bar' as varchar(20)) as varchar(30))", createVarcharType(30), "bar");
+        assertFunction("cast(cast('bar' as varchar(20)) as varchar)", VARCHAR, "bar");
+
+        assertFunction("cast('banana' as varchar(3))", createVarcharType(3), "ban");
+        assertFunction("cast(cast('banana' as varchar(20)) as varchar(3))", createVarcharType(3), "ban");
+    }
+
+    @Test
+    public void testVarcharToCharCast()
+    {
+        assertFunction("cast('bar  ' as char(10))", createCharType(10), "bar");
```

Comment:
`assertFunction` verifies internal Presto representation of a value. The test is correct since we represent `CHAR` without trailing spaces.

Having `CHAR` represented internally with trailing spaces wouldn't make much sense. Such values would require more memory. It is also harder to add spaces than to trim them (e.g. when reading data from Hive). Adding spaces requires counting code points.